webapp/src/app.tsx

```diff
@@ -1283,10 +1283,19 @@ export class ProjectView
     setSideFile(fn: pkg.File, line?: number) {
         let fileName = fn.name;
         let currFile = this.state.currFile.name;
+
         if (fileName != currFile && pxt.editor.isBlocks(fn)) {
-            // Going from ts -> blocks
+            // Going from ts/py -> blocks
             pxt.tickEvent("sidebar.showBlocks");
             this.openBlocks();
+        } else if (fileName.endsWith(".py") && currFile.endsWith(".ts")) {
```

Comment:
do we need to special case ts->py / py->ts here vs. just 'going from not py -> py' / 'going from not ts -> ts'? I might be missing something but it feels like there'd be edge cases with 'ts -> json/jres file -> py' here like the repro steps if it's specifically checking curr file & newly set file?

Reply:
This turned out to be somewhat involved, as there are quite a few places in the language-switching code that assume you have main.* selected. I went in and tried to update the critical locations to remove this assumption, so I think it should be working now and my own testing seems to have gone okay, but totally possible I missed something.

New upload target: https://arcade.makecode.com/app/962a0b823b61325c162a2fed0995f02a87f2aec4-d0020cc65b